src/coprocessor/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,392 @@
+    }
+
+    pub fn cast_decimal_as_time(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_decimal_as_duration(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_int(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_real(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_string(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_decimal(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_time(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_string_as_duration(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_int(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_real(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_string(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_decimal(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_time(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_time_as_duration(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
+        // TODO: add impl
+        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    }
+
+    pub fn cast_duration_as_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        if let Datum::Dur(_) = d {
+            let i = try!(try!(d.into_dec()).as_i64().into_result());
```

Comment:
Add the call to `RoundFrac`, refer to
https://github.com/pingcap/tidb/blob/master/expression/builtin_cast.go#L846